ingest/scripts/compare_hashes.py

```diff
@@ -215,6 +231,8 @@ def main(
                 if record[key]
             ]
         )
+        insdc_accessions = [record[key] for key in insdc_keys if record.get(key)]
+        ingested_insdc_accessions.difference_update(insdc_accessions)
```

Comment:
Here we should create a new variable with a good name rather than mutate, makes it easier to review 😀

Reply:
Oh I created the `ingested_insdc_accessions` variable just to mutate it - should I maybe choose a different name so that that is clearer? 

ingested_insdc_accessions_not_yet_reingested?

Reply:
I think the idea would be `result = ingested_insdc_accessions.difference(insdc_accessions)` (or something)

Reply:
but I update the variable ingested_insdc_accessions on every iteration over the new entries - after the first difference I would need to take the difference from the potentially_revoked list

Reply:
An alternative would be to create two sets and check their difference
```
missing_records = previous_records - set(new_records)
if missing_records:
```

Reply:
Yep, sorry, `ingested_insdc_accessions_not_yet_reingested` makes sense to me.

Reply:
> An alternative would be

That's also great - and probably easier to follow

Reply:
Ok I changed to use 2 lists now (and made a couple small changes to make ruff happy ;-) ) - is this clearer now?

Reply:
@corneliusroemer is it ok if I merge?

Reply:
Merging as @corneliusroemer told me offline that was ok